internal/git/tag.go

```diff
@@ -109,31 +101,72 @@ func TagsFromLocal(repoPath string) ([]Tag, error) {
 		return nil, err
 	}
 
-	tagrefs, err := r.Tags()
+	tagRefs, err := r.Tags()
 	if err != nil {
 		return nil, err
 	}
 
 	var tags []Tag
 	for {
-		t, err := tagrefs.Next()
+		t, err := tagRefs.Next()
 		if err == io.EOF || t == nil {
 			break
 		} else if err != nil {
 			return nil, err
 		}
 
-		c, err := r.CommitObject(t.Hash())
+		tag, err := newTag(r, t)
 		if err != nil {
-			log.Debugf("unable to get tag '%s' info from commit=%q: %w", t.Name().String(), t.Hash().String(), err)
+			// TODO
+			return nil, err
+		}
+		if tag == nil {
 			continue
 		}
 
-		tags = append(tags, Tag{
+		tags = append(tags, *tag)
+	}
+	return tags, nil
+}
+
+func newTag(r *git.Repository, t *plumbing.Reference) (*Tag, error) {
+	// the plumbing reference is to the tag. For a lightweight tag, the tag object points directly to the commit
+	// with the code blob. For an annotated tag, the tag object has a commit for the tag itself, but resolves to
+	// the commit with the code blob. It's important to use the timestamp from the tag object when available
+	// for annotated tags and to use the commit timestamp for lightweight tags.
+
+	if !t.Name().IsTag() {
+		return nil, nil
+	}
+
+	c, err := r.CommitObject(t.Hash())
+	if err == nil && c != nil {
+		// this is a lightweight tag... the tag hash points directly to the commit object
+		return &Tag{
 			Name:      t.Name().Short(),
 			Timestamp: c.Committer.When,
-			Commit:    t.Hash().String(),
-		})
+			Commit:    c.Hash.String(),
+			Annotated: false,
+		}, nil
 	}
-	return tags, nil
+
+	// this is an annotated tag... the tag hash points to a tag object, which points to the commit object
+	// use the timestamp info from the tag object
+
+	tagObj, err := object.GetTag(r.Storer, t.Hash())
+	if err != nil {
+		return nil, fmt.Errorf("unable to resolve tag for %q: %w", t.Name(), err)
+	}
+
+	if tagObj == nil {
+		return nil, fmt.Errorf("unable to resolve tag for %q", t.Name())
+	}
+
+	return &Tag{
+		Name: t.Name().Short(),
+		// without the timezone info the timezone will have zone info as the offset, which is already encoded...
+		Timestamp: tagObj.Tagger.When.In(time.Local),
```

Comment:
I know the comment is explaining why this isn't UTC, but I still don't understand why this isn't UTC.